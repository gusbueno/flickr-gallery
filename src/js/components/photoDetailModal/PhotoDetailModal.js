'use strict'

import React from 'react'
import PropTypes from 'prop-types'

import styles from './PhotoDetailModal.scss'

const PhotoDetailModal = ({ photo, onClosePhotoDetail }) => {
  return (
    <div className={styles['photo-detail-modal']}>
      <div className={styles['photo-header']}>
        <span className={styles['photo-title']}>{photo.title._content}</span>
        <span className={styles['close']} onClick={() => onClosePhotoDetail()}>Close</span>
      </div>
      <div className={styles['photo-wrapper']}>
        <img src={`https://farm${photo.farm}.staticflickr.com/${photo.server}/${photo.id}_${photo.secret}.jpg`} alt={photo.title._content} />
      </div>
      <div className={styles['photo-info']}>
        <div className={styles['info-group']}>
          <span className={styles['label']}>Owner: </span>
          <span className={styles['value']}>{photo.owner.username}</span>
        </div>

        <div className={styles['info-group']}>
          <span className={styles['label']}>Date taken: </span>
          <span className={styles['value']}>{photo.dates.taken}</span>
        </div>

        <div className={styles['info-group']}>
          <span className={styles['label']}>Views: </span>
          <span className={styles['value']}>{photo.views}</span>
        </div>

        <div className={styles['info-group']}>
          <span className={styles['label']}>Comments: </span>
          <span className={styles['value']}>{photo.comments._content}</span>
        </div>
        <a href={photo.urls.url[0]._content} target='blank'>Full size</a>
      </div>
    </div>
  )
}

PhotoDetailModal.propTypes = {
  photo: PropTypes.object.isRequired,
  onClosePhotoDetail: PropTypes.func.isRequired
}

export default PhotoDetailModal
