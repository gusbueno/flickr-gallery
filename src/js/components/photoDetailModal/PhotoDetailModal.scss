.photo-detail-modal {
  display: flex;
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  background-color: rgba(white, 0.9);
  padding: 20px 50px;
  flex-direction: column;
  box-sizing: border-box;

  .photo-header {
    display: flex;
    justify-content: space-between;
    .photo-title {
      font-size: 24px;
      font-weight: bold;
      margin-bottom: 20px;
      color: #373F51;
    }

    .close {
      color: #3F70E2;
      font-weight: bold;
      &:hover {
        cursor: pointer;
      }
    }
  }
  

  .photo-wrapper {
    height: 400px;
    width: 100%;
    display: flex;
    justify-content: center;
    overflow: hidden;
    margin-bottom: 20px;

    img {
      height: 100%;
      border-radius: 5px;
    }
  }

  .photo-info {
    display: flex;
    justify-content: space-between;

    .info-group {
      .label {
        font-size: 12px;
        color: #373F51;
      }

      .value {
        font-size: 16px;
        color: #373F51;
        font-weight: bold;
      }
    }
  }
}