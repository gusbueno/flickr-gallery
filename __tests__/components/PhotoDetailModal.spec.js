/* global describe, it, expect, jest */

import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import PhotoDetailModal from '../../src/js/components/photoDetailModal/PhotoDetailModal'

describe('<PhotoDetailModal /> Component', () => {
  const photo = {
    title: { _content: 'Title' },
    farm: 123,
    server: '123',
    id: '123',
    secret: '123',
    owner: { username: 'Pepe' },
    dates: { taken: '20/01/2018' },
    views: 12,
    comments: { _content: 12 },
    urls: {
      url: [{ _content: 'https://awesomeurl.com/my/photo' }]
    }
  }
  it('should match with the snapshot', () => {
    const tree = renderer.create(
      <PhotoDetailModal photo={photo} onClosePhotoDetail={() => {}} />
    )
    const json = tree.toJSON()
    expect(json).toMatchSnapshot()
  })

  const onClosePhotoDetailMock = jest.fn()
  const photoDetailModalShallow = shallow(
    <PhotoDetailModal photo={photo} onClosePhotoDetail={onClosePhotoDetailMock} />
  )

  it('should exists', () => {
    expect(photoDetailModalShallow.length).toEqual(1)
  })

  it('should render pagination wrapper', () => {
    expect(photoDetailModalShallow.find('.photo-detail-modal').length).toEqual(1)
  })

  it('should call onClosePhotoDetail on click close button', () => {
    photoDetailModalShallow.find('.close').first().simulate('click')
    expect(onClosePhotoDetailMock).toHaveBeenCalled()
  })
})
