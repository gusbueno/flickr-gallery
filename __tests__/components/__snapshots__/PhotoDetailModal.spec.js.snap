// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoDetailModal /> Component should match with the snapshot 1`] = `
<div
  className="photo-detail-modal"
>
  <div
    className="photo-header"
  >
    <span
      className="photo-title"
    >
      Title
    </span>
    <span
      className="close"
      onClick={[Function]}
    >
      Close
    </span>
  </div>
  <div
    className="photo-wrapper"
  >
    <img
      alt="Title"
      src="https://farm123.staticflickr.com/123/123_123.jpg"
    />
  </div>
  <div
    className="photo-info"
  >
    <div
      className="info-group"
    >
      <span
        className="label"
      >
        Owner: 
      </span>
      <span
        className="value"
      >
        Pepe
      </span>
    </div>
    <div
      className="info-group"
    >
      <span
        className="label"
      >
        Date taken: 
      </span>
      <span
        className="value"
      >
        20/01/2018
      </span>
    </div>
    <div
      className="info-group"
    >
      <span
        className="label"
      >
        Views: 
      </span>
      <span
        className="value"
      >
        12
      </span>
    </div>
    <div
      className="info-group"
    >
      <span
        className="label"
      >
        Comments: 
      </span>
      <span
        className="value"
      >
        12
      </span>
    </div>
    <a
      href="https://awesomeurl.com/my/photo"
      target="blank"
    >
      Full size
    </a>
  </div>
</div>
`;
